<a href="https://colab.research.google.com/github/rishav-raj-genx/Data_Science_26/blob/main/Basic_synatx_and_some_questions_DAY_0%2B1%2B2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
print("Hello World!")

Hello World!


In [ ]:
print("Hello Machines...")

Hello Machines...


In [ ]:
a = 10
print(a * 2 + 5)

25


In [ ]:
a = 5
b = 10
c = 15
d = 20
e = 25
print(a+b+c+d+e)

75


In [ ]:
a = input()
b = input()
c = input()
print(a + b + c)

2
3
4
234


In [ ]:
a = input()
b = input()
c = input()
print(a+b+c)

1
2
3
123


In [ ]:
a = 1 * 10**9
b = 1 * 10**8
print(a+b)

1100000000


In [ ]:
a = 1
b = 5
# print(a + b + c)

In [ ]:
a = 10
b = 20
a, b = b , a
print (a, b)

20 10


In [ ]:
for i in range(1, 51):
    if i % 2 == 0:
        print("Even no :", i)
    else:
        print("Odd no :", i)


Odd no : 1
Even no : 2
Odd no : 3
Even no : 4
Odd no : 5
Even no : 6
Odd no : 7
Even no : 8
Odd no : 9
Even no : 10
Odd no : 11
Even no : 12
Odd no : 13
Even no : 14
Odd no : 15
Even no : 16
Odd no : 17
Even no : 18
Odd no : 19
Even no : 20
Odd no : 21
Even no : 22
Odd no : 23
Even no : 24
Odd no : 25
Even no : 26
Odd no : 27
Even no : 28
Odd no : 29
Even no : 30
Odd no : 31
Even no : 32
Odd no : 33
Even no : 34
Odd no : 35
Even no : 36
Odd no : 37
Even no : 38
Odd no : 39
Even no : 40
Odd no : 41
Even no : 42
Odd no : 43
Even no : 44
Odd no : 45
Even no : 46
Odd no : 47
Even no : 48
Odd no : 49
Even no : 50


In [ ]:
a = 2
if a % 2 == 0:
  print("Even")
else :
  print("Odd")

Even


In [ ]:
a = 0
if a > 0:
  print("Positive")
elif a == 0:
  print("Zero")
else :
  print("Negative")

Zero


In [ ]:
a = input()
if a == "Sun":
  print("Sunday")
elif a == "Mon":
  print("Monday")
elif a == "Tue":
  print("Tuesday")
else :
  print("Wednesday")

tue
Wednesday


In [ ]:
for i in range(0, 11, 2):
  print(i)

0
2
4
6
8
10


In [ ]:
for i in range(1, 11, 2):
  print(i)

1
3
5
7
9


In [ ]:
a = 0
while a < 10:
  print(a)
  a = a+1

0
1
2
3
4
5
6
7
8
9


In [ ]:
s = input()
rev = ""
for ch in s:
  rev = ch + rev
if s == rev:
  print("Pallindrome")
else :
  print("Not A Pallindrome")

HomeemoH
Pallindrome


In [ ]:
lst = [1, 2, 3, "Hi", 'Hi', 'true', 'false', 'c', [2,3]]
print(
    lst
)

[1, 2, 3, 'Hi', 'Hi', 'true', 'false', 'c', [2, 3]]


Array can only store same type of datatype

While List can store heterogeneous type of data

we fetch array from numpy

and list takes time to execute

In [ ]:
lo = [182, 81, 'yo', 'yu', [8812, [199, 188], 881]]
lo.append(2)
print(lo)
lo.append([82, [818, 716,1445]])
print(lo)
lo.extend([2, [881, [7612, [9182, 61552]]], 7612, 'true'])
print(lo)
lo.insert(2, 'false')
print(lo)

#append - adds element at last - every value
#extend - to add multiple elements in our list, requires an iterable data structure (at last), and seperates all the elements
#insert - to add element at a custom value of index - in between



[182, 81, 'yo', 'yu', [8812, [199, 188], 881], 2]
[182, 81, 'yo', 'yu', [8812, [199, 188], 881], 2, [82, [818, 716, 1445]]]
[182, 81, 'yo', 'yu', [8812, [199, 188], 881], 2, [82, [818, 716, 1445]], 2, [881, [7612, [9182, 61552]]], 7612, 'true']
[182, 81, 'false', 'yo', 'yu', [8812, [199, 188], 881], 2, [82, [818, 716, 1445]], 2, [881, [7612, [9182, 61552]]], 7612, 'true']


In [ ]:
ok = [20]
ok.append('hi')
ok.extend(['everyone', 'how is life going'])
ok.insert(1, 'Good Morning and')
print(ok)

[20, 'Good Morning and', 'hi', 'everyone', 'how is life going']


In [ ]:
# Remove function from a list - pop
#pop - remove from the end
#remove - requires value not index and returns null
#index - returns index of the value
#count - tells how many times an element is in an index
#sort - sorts the list (lexographically in string case) - preference of homogenous list
lo = [182, 81, 'yo', 'yu', [8812, [199, 188], 881], 91, 81, 21]
lo.pop()
print(lo)
lo.remove(182)
print(lo)
print(lo.index(81))
print(lo.count(81))
op = [9191, 7199,6457,123456,987654,7623,2,18731,981342]
op.sort()
print(op)

[182, 81, 'yo', 'yu', [8812, [199, 188], 881], 91, 81]
[81, 'yo', 'yu', [8812, [199, 188], 881], 91, 81]
0
2
[2, 6457, 7199, 7623, 9191, 18731, 123456, 981342, 987654]


In [ ]:
lol = [12, 34, 56, 78, 910, -109, -87, -65, -43, -21, 0]
print(sum(lol))
pole = [6, 78, 910, -109, -8]
print(min(pole))
print(max(pole))
print(sorted(pole))
#sort - is a method
#sorted - is a function in which list goes as a parameter
#sum - returns sum of the list
#min - smallest element return
#max - largest element return

765
-109
910
[-109, -8, 6, 78, 910]


In [ ]:
lst = [-66, 56, 42, 907, 235]
a = sorted(lst)
print(a)

[-66, 42, 56, 235, 907]


In [ ]:
#lists that are immutable
tp = (21,93,56,74,80)
print(tp)
u = sorted(tp)
print(u)
#sorted function returns a list that are changable (mutable) hence the answer we get on using sorted function on a tuple is a list

(21, 93, 56, 74, 80)
[21, 56, 74, 80, 93]


Tuple Functions

count values - count()

find - index()

size - len()

total - sum()

highest/lowest - max()/min()

sort - sorted() [returns list]

In [ ]:
Tup = (20,43,95,76,81,20)
print(Tup.count(20)) #count - used to tell how many times a number returns
print(Tup.index(95)) #index - used to find the index of a value provided
print(len(Tup)) #len - tells the length of the tuple
print(sum(Tup)) #sum - sum of all the elements in the tuple
print(max(Tup)) #max - returns the maximum element
print(min(Tup)) #min - returns the maximum element
print(sorted(Tup)) #sorted - returns list after sorting a tuple

2
2
6
335
95
20
[20, 20, 43, 76, 81, 95]


In [ ]:
l1 = []
l2 = []
size = int(input()) #size is to be given in one digits - don't forget
for i in range(size):
  x = int(input())
  l1.append(x)
for i in range(size):
  y = int(input())
  l2.append(y)
l3 = []
for i in range(size):
  l3.append(l1[i] + l2[i])
print(l3)
  # l3[i] = l1[i] + l2[i] - index error since l3 meine koi element hai hi nhi
  # print(l3)

3
2
7
9
2
6
4
[4, 13, 13]


# input should be given in different line like - 1
#                                           2
  #                                             3
not like 123
or 1 2 3

n = int(input())

answer = []

for i in range (n):

    x = int(input())

    answer.append(x)

    if (answer[i] % 3 == 0 and answer[i] % 5 == 0):

        answer[i] = "FizzBuzz"

    elif (answer[i] % 3 == 0):

        answer[i] = "Fizz"

    elif (answer[i] % 5 == 0):

        answer[i] = "Buzz"

    else:
    
        answer[i] = str(answer[i])

print(answer)

#Although the code is correct we need to check i not answer[i] and no input taking in leetcode since it already has input feeded in it already so the correct is below

In [ ]:
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        answer = []
        for i in range (1, n + 1):
            if i % 3 == 0 and i % 5 == 0:
                    answer.append("FizzBuzz")
            elif i % 3 == 0:
                    answer.append("Fizz")
            elif i % 5 == 0:
                    answer.append("Buzz")
            else:
                    answer.append(str(i))
        return answer
#412 Fizz Buzz

In [ ]:
def fun_name(parameters):

In [ ]:
a = int(input())
b = int(input())
def add(a, b):
  c = a+b
  print(c)
add(a,b)

329
678
1007


In [ ]:
def add_(a,b):
  return (a + b)

In [ ]:
#lambda function
x = lambda a : 2 * a
print(x(2))

4


##Strings


In [ ]:
st = ""
print(len(st))
st = "anand"
st1 = st.upper()
print(st1)
st2 = st.lower()
print(st2)
st3 = st.capitalize()
print(st3)
st4 = st.title()
print(st4)
# upper - capitalizes all the letters in the string
# len - tells the length of the strings
# lower - small the letters of the string
# capitalize -

In [ ]:
st = ""
print(len(st))
st = "Joseph Lee"
st1 = st.upper()
print(st1)
st2 = st.lower()
print(st2)
st3 = st.capitalize()
print(st3)
st4 = st.title()
print(st4)
print(len(st))

0
JOSEPH LEE
joseph lee
Joseph lee
Joseph Lee
10


In [ ]:
f_name = ' Ivanov '
m_name = ' Elijah '
l_name = ' Petrov '
print(f_name.strip() + m_name.strip() + l_name.strip())
print(f_name.lstrip() + ' ' + m_name.strip() + ' ' + l_name.strip())
new_l_name = ' Lee Lao'
print(f_name.strip() + m_name.strip() + new_l_name.rstrip())
#strip - removes spaces from both left and right
#rstrip - removes spaces from right side
#lstrip - remove spaces from the left side

IvanovElijahPetrov
Ivanov  Elijah Petrov
IvanovElijah Lee Lao


In [ ]:
a = ' today '
b = " weather "
c = "is GooD   " #only GooD not good or GOOD or gOOd
d = c.lower()
print(a.title() + ',' + b.rstrip() + ' ' + d.rstrip() + '.')
#or
print(a.title() + ',' + b.rstrip() + ' ' + c.lower().rstrip() + '.')
print(c.replace('GooD', 'pleasant'))
ans = a.strip().capitalize(), ", " + b.rstrip(), c.replace('GooD','Pleasant').lower().rstrip(), "."
print(ans)



 Today , weather is good.
 Today , weather is good.
is pleasant   
('Today', ',  weather', 'is pleasant', '.')


In [ ]:
a = " today "
b = " weather "
c = "is GooD   "
ans = a.strip().capitalize() +  ", " + b.strip() + " " + c.lower().strip() + "."
print(ans)
print(ans.find("Today"))
print(ans.find("good"))
print(ans.startswith("Today"))
print(ans.endswith("good"))

Today, weather is good.
0
18
True
False


In [ ]:
s = "Panda"
rev = ""
for i in s:
  rev = i + rev
print(rev)

adnaP


In [ ]:
s = "Hello Everyone, Good Morning"
print(s[-7])
print(s[1 : 9])
print(s[-16 : -4])
print(s[4 ::])
print(s[:: -3]) #reverses the string
print(s[:3])
print(s[:9])
#This is called slicing not spclicing in Python

M
ello Eve
ne, Good Mor
o Everyone, Good Morning
gnMo nrElH
Hel
Hello Eve


In [ ]:
set = {1,827.818, 7162,7176,882,982,77,77, 1629.928}
print(set)

{1, 7176, 77, 882, 982, 7162, 827.818, 1629.928}


In [ ]:
dict = {
    "Name" : ["Ivan", "Elijah", "Sergei", "Roman"],
    "Age" : ["29", "30", "24", "25"],
    "Occupation" : "Data Scientist",
    "Salary" : "$120K",}
print(dict)
print(dict['Name'])
print(dict["Name"][2])
dict['id'] = 12
print(dict)
del dict["Occupation"]
print(dict)

{'Name': ['Ivan', 'Elijah', 'Sergei', 'Roman'], 'Age': ['29', '30', '24', '25'], 'Occupation': 'Data Scientist', 'Salary': '$120K'}
['Ivan', 'Elijah', 'Sergei', 'Roman']
Sergei
{'Name': ['Ivan', 'Elijah', 'Sergei', 'Roman'], 'Age': ['29', '30', '24', '25'], 'Occupation': 'Data Scientist', 'Salary': '$120K', 'id': 12}
{'Name': ['Ivan', 'Elijah', 'Sergei', 'Roman'], 'Age': ['29', '30', '24', '25'], 'Salary': '$120K', 'id': 12}


In [ ]:
#Explore functions of sets and dictonaries

#Sets Method -

In [ ]:
Set1 = {1,23,32,87,716}
Set1.add(81)
print(Set1)
#add() - add element to the set

In [ ]:
Set2 = {187,816,918.816, 61662,661,6525,1552,524,8775,7781,971}
Set2.clear()
print(Set2)
#Set2 - Remove all the elements from the set

set()


In [ ]:
s = {772,7661,51525,312,991,6871,5671,872,12}
t = s.copy()
print(t)
print(s)
#copy() - returns copy of a set

{772, 51525, 5671, 872, 12, 7661, 6871, 312, 991}
{772, 51525, 5671, 872, 12, 7661, 6871, 312, 991}


In [ ]:
s = {1,2,3,4,5,6,7,8,9,10}
t = {1,2,3,4,6,7}
u = s.difference(t)
print(u)
#difference() - returns a set containing the difference between two or more sets - remaining elements of the set we subtracted a different set from.

{8, 9, 10, 5}


In [ ]:
s = {22,33,44,55,66,67,87}
t = {77,67,66,515,627}
s.difference_update(t)
print(s)

a = {"apple", "banana", "cherry"}
b = {"google", "microsoft", "apple"}

a -= b

print(a)
#difference_update() - Removes the items in this set that are also included in another, specified set

{33, 22, 55, 87, 44}
{'cherry', 'banana'}


In [ ]:
s = {1,2,3,4,5,6,7,8,9,10}
t = {1,2,3,4,6,7}
u = s.intersection(t)
print(u)
x = {"a", "b", "c"}
y = {"c", "d", "e"}
z = {"f", "g", "c"}
result = x & y & z
print(result)
#intersection() - Return a set that contains the items that exist in both set x, and set y:

{1, 2, 3, 4, 6, 7}


In [ ]:
s = {1,2,3,4,5,6,7,8,9,10}
t = {1,2,3,4,6,7}
s.intersection_update(t)
print(s)
x = {"apple", "banana", "cherry"}
y = {"google", "microsoft", "apple"}
x &= y
print(x)
#intersection_update() - Remove the items that is not present in both x and y:

{1, 2, 3, 4, 6, 7}
{'apple'}


In [ ]:
s = {1,2,3,4,5,6,7,8,9,10}
t = {1,2,3,4,6,7}
u = s.isdisjoint(t)
print(u)
#isdisjoint() - The isdisjoint() method returns True if none of the items are present in both sets, otherwise it returns False.

False


In [ ]:
fruits = {"apple", "banana", "cherry"}
fruits.discard("banana")
print(fruits)
#discard() - The discard() method removes the specified item from the set.

{'apple', 'cherry'}


In [ ]:
s = {1,2,3,4,5,6,7,8,9,10}
t = {1,2,3,4,6,7}
u = s.issubset(t)
print(u)
#issubset() - The issubset() method returns True if all items in the set exists in the specified set, otherwise it returns False.
#Use <= as a shortcut instead of issubset()

In [ ]:
x = {"f", "e", "d", "c", "b", "a"}
y = {"a", "b", "c"}
z = x.issuperset(y)
print(z)
#issuperset() - Returns True if all items of another set is present in this set
#you can use the >= operator instead

In [ ]:
fruits = {"apple", "banana", "cherry"}
fruits.pop()
print(fruits)
#pop() - Removes an element from the set

{'cherry', 'banana'}


In [ ]:
fruits = {"apple", "banana", "cherry"}
fruits.remove("banana")
print(fruits)
#remove()	-	Removes the specified element

In [ ]:
x = {"apple", "banana", "cherry"}
y = {"google", "microsoft", "apple"}
z = x.symmetric_difference(y)
print(z)
#symmetric_difference()	-	Returns a set with the symmetric differences of two sets (^)

In [ ]:
x = {"apple", "banana", "cherry"}
y = {"google", "microsoft", "apple"}
x.symmetric_difference_update(y)
print(x)
# symmetric_difference_update()	- [^=]Inserts the symmetric differences from this set and another

In [ ]:
x = {"apple", "banana", "cherry"}
y = {"google", "microsoft", "apple"}
z = x.union(y)
print(z)
#union() - can use this symbol too [|]. Return a set containing the union of sets

In [ ]:
s = {1,2,3,4,5,6,7,8,9,10}
t = {1,2,3,4,6,7,11}
u = s.update(t)
print(s)
#update() - Update the set with the union of this set and others

{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}


#Dictonary Methods -
Method	Description

clear()	Removes all the elements from the dictionary

copy()	Returns a copy of the dictionary

fromkeys()	Returns a dictionary with the specified keys and value

get()	Returns the value of the specified key

items()	Returns a list containing a tuple for each key value pair

keys()	Returns a list containing the dictionary's keys

pop()	Removes the element with the specified key

popitem()	Removes the last inserted key-value pair

setdefault()	Returns the value of the specified key. If the key does not exist: insert the key, with the specified value

update()	Updates the dictionary with the specified key-value pairs

values()	Returns a list of all the values in the dictionary


In [ ]:
car = {
  "brand": "Ford",
  "model": "Mustang",
  "year": 1964
}
car.clear()
print(car)

{}


In [ ]:
text = {
    "language" : "Cantonese",
    "font-style" : "CurlzMT"
}
text1 = text.copy()
print(text1)

{'language': 'Cantonese', 'font-style': 'CurlzMT'}


In [ ]:
text = {
    "language" : "Cantonese",
    "font-style" : "CurlzMT"
}
x = ('korean', 'Times Roman', "Hi")
text1 = dict.fromkeys(text, x)
print(text1)

{'language': ('korean', 'Times Roman', 'Hi'), 'font-style': ('korean', 'Times Roman', 'Hi')}


In [ ]:
text = {
    "language" : "Cantonese",
    "font-style" : "CurlzMT"
}
x = text.get("language")
print(x)


Cantonese


In [ ]:
text = {
    "language" : "Cantonese",
    "font-style" : "CurlzMT"
}
x = text.items()
print(x)


dict_items([('language', 'Cantonese'), ('font-style', 'CurlzMT')])


In [ ]:
car = {
    "brand": "Ford",
  "model": "Mustang",
  "year": 1964
}
x = car.keys()
print(x)


dict_keys(['brand', 'model', 'year'])


In [ ]:
car = {
    "brand": "Ford",
  "model": "Mustang",
  "year": 1964
}
car.pop("model")
print(car)

{'brand': 'Ford', 'year': 1964}


In [ ]:
text = {
    "language" : "Cantonese",
    "font-style" : "CurlzMT"
}
text.popitem()
print(text)


In [ ]:
car = {
    "brand": "Ford",
  "model": "Mustang",
  "year": 1964
}
x = car.setdefault("model", "Bronco")
print(x)


Mustang


In [ ]:
text = {
    "language" : "Cantonese",
    "font-style" : "CurlzMT"
}
text.update({"font-style" : "Times Roman"})
print(text)


{'language': 'Cantonese', 'font-style': 'Times Roman'}


In [ ]:
text = {
    "language" : "Cantonese",
    "font-style" : "CurlzMT"
}
x = text.values()
print(x)


dict_values(['Cantonese', 'CurlzMT'])
